ui/packages/consul-ui/tests/integration/adapters/nspace-test.js

```diff
@@ -6,14 +6,17 @@ module('Integration | Adapter | nspace', function(hooks) {
   setupTest(hooks);
   const id = 'slug';
   const dc = 'dc-1';
-  test('requestForQuery returns the correct url/method', function(assert) {
+  test('requestForQuery returns the correct url/method', async function(assert) {
     const adapter = this.owner.lookup('adapter:nspace');
     const client = this.owner.lookup('service:client/http');
-    const request = client.requestParams.bind(client);
+    const request = function() {
+      return () => client.requestParams.bind(client)(...arguments);
+    };
     const expected = `GET /v1/namespaces?dc=${dc}`;
-    const actual = adapter.requestForQuery(request, {
+    let actual = await adapter.requestForQuery(request, {
       dc: dc,
     });
+    actual = actual();
```

Comment:
The work here was just because we had to change the shape of our test mocking as we started using `respond` in the adapters method that we added above in this git changeset. We used the exact same approach as we do in the equivalent partition test:

https://github.com/hashicorp/consul/blob/20cc028ac02af77ffff2bde19939c1c4fab37c1a/ui/packages/consul-ui/tests/integration/adapters/partition-test.js#L24-L32

Just to note, we don't test anything extra here in this PR, this was just to get the previous test passing again due to the shape change.